src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -139,12 +142,22 @@ private void InitializeCore()
         }
 
         private void UpdateStateMachine()
+        {
+            ThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                await UpdateStateMachineAsync();
+            });
+        }
+
+        private async Task UpdateStateMachineAsync()
         {
             AssertIsForeground();
```

Comment:
if you are using JTF, don't use AssetIsForeground() but just switch to main thread yourself

ThreadHelper.SwitchToMainThread…

also, rather than using ThreadHelper directly, you might want to use _threadContext thing @sharwell added by importing it in MEF constructor.